test/test_unistd.rs

```diff
@@ -139,46 +144,47 @@ macro_rules! execve_test_factory(
                 assert!(string.contains("baz=quux"));
             }
         }
+        drop(m);    // appease the unused_variable checker
     }
     )
 );
 
 #[test]
 fn test_fchdir() {
+    // fchdir changes the process's cwd
+    let m = ::CWD_MTX.lock().expect("Mutex got poisoned by another test");
     let tmpdir = TempDir::new("test_fchdir").unwrap();
     let tmpdir_path = tmpdir.path().canonicalize().unwrap();
     let tmpdir_fd = File::open(&tmpdir_path).unwrap().into_raw_fd();
-    let olddir_path = getcwd().unwrap();
-    let olddir_fd = File::open(&olddir_path).unwrap().into_raw_fd();
 
     assert!(fchdir(tmpdir_fd).is_ok());
     assert_eq!(getcwd().unwrap(), tmpdir_path);
 
-    assert!(fchdir(olddir_fd).is_ok());
-    assert_eq!(getcwd().unwrap(), olddir_path);
-
-    assert!(close(olddir_fd).is_ok());
     assert!(close(tmpdir_fd).is_ok());
+    drop(m);    // appease the unused_variable checker
 }
 
 #[test]
 fn test_getcwd() {
+    // chdir changes the process's cwd
+    let m = ::CWD_MTX.lock().expect("Mutex got poisoned by another test");
     let tmp_dir = TempDir::new("test_getcwd").unwrap();
     assert!(chdir(tmp_dir.path()).is_ok());
-    assert_eq!(getcwd().unwrap(), current_dir().unwrap());
+    assert_eq!(getcwd().unwrap(), tmp_dir.path());
 
-    // make path 500 chars longer so that buffer doubling in getcwd kicks in.
-    // Note: One path cannot be longer than 255 bytes (NAME_MAX)
-    // whole path cannot be longer than PATH_MAX (usually 4096 on linux, 1024 on macos)
+    // make path 500 chars longer so that buffer doubling in getcwd
+    // kicks in.  Note: One path cannot be longer than 255 bytes
+    // (NAME_MAX) whole path cannot be longer than PATH_MAX (usually
+    // 4096 on linux, 1024 on macos)
     let mut inner_tmp_dir = tmp_dir.path().to_path_buf();
     for _ in 0..5 {
         let newdir = iter::repeat("a").take(100).collect::<String>();
-        //inner_tmp_dir = inner_tmp_dir.join(newdir).path();
         inner_tmp_dir.push(newdir);
         assert!(mkdir(inner_tmp_dir.as_path(), stat::S_IRWXU).is_ok());
     }
     assert!(chdir(inner_tmp_dir.as_path()).is_ok());
-    assert_eq!(getcwd().unwrap(), current_dir().unwrap());
+    assert_eq!(getcwd().unwrap(), inner_tmp_dir.as_path());
+    drop(m);    // appease the unused_variable checker
```

Comment:
I didn't know you could apply that to a single variable.  Cool!  BTW, if you didn't figure out, it doesn't work to do "let _ = ..." because the anonymous variable gets dropped right away.